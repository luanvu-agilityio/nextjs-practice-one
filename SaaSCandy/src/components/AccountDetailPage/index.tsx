'use client';

import { Fragment, useEffect } from 'react';
import Link from 'next/link';
import { useSession, signOut } from 'next-auth/react';
import { useRouter } from 'next/navigation';

// Components
import { Button, Heading, Typography } from '@/components/common';
import {
  Breadcrumb,
  BreadcrumbList,
  BreadcrumbItem,
  BreadcrumbLink,
  BreadcrumbSeparator,
  BreadcrumbPage,
} from '@/components/common/Breadcrumb';
import { Spinner } from '../ui/spinner';

// Icons
import LogoIcon from '@/components/icons/Logo';

// Utils & Constants
import { ROUTES } from '@/constants';

function AccountDetailPage() {
  const { data: session, status } = useSession();
  const router = useRouter();

  // Redirect if not authenticated
  useEffect(() => {
    if (status === 'unauthenticated') {
      router.push(ROUTES.SIGN_IN);
    }
  }, [status, router]);

  if (status === 'loading') {
    return (
      <div className='flex justify-center items-center min-h-screen'>
        <Spinner />
      </div>
    );
  }

  if (!session?.user) {
    return null;
  }

  const {
    user: { name, email, image },
  } = session;

  const handleGoBack = () => {
    router.back();
  };

  const handleSignOut = async () => {
    try {
      await signOut({
        callbackUrl: ROUTES.HOME,
      });
    } catch (error) {
      console.error('Sign out error:', error);
    }
  };

  const breadcrumbs = [
    { label: 'Home', href: '/', isActive: false },
    { label: 'Account', href: '/account', isActive: true },
  ];

  return (
    <div className='flex flex-col items-center max-w-[1296px] mx-auto min-h-screen py-8'>
      {/* Header Section */}
      <div className='pb-12 w-full'>
        <Heading
          as='h1'
          size='xl'
          content='Account Details'
          className='text-center'
        />

        {/* Breadcrumb */}
        <Breadcrumb className='justify-center mt-4'>
          <BreadcrumbList>
            {breadcrumbs.map((item, index) => (
              <Fragment key={item.label}>
                <BreadcrumbItem>
                  {item.isActive ? (
                    <BreadcrumbPage>{item.label}</BreadcrumbPage>
                  ) : (
                    <BreadcrumbLink href={item.href}>
                      {item.label}
                    </BreadcrumbLink>
                  )}
                </BreadcrumbItem>
                {index < breadcrumbs.length - 1 && <BreadcrumbSeparator />}
              </Fragment>
            ))}
          </BreadcrumbList>
        </Breadcrumb>
      </div>

      {/* Account Detail Section */}
      <div className='py-14 px-16 w-full max-w-159 rounded-4xl border border-form-border-color shadow-form bg-white'>
        <div className='flex flex-col gap-10'>
          {/* Logo */}
          <div className='flex justify-center'>
            <Link href={ROUTES.HOME} className='flex items-center gap-3'>
              <LogoIcon className='w-10 h-10' />
              <span className='text-3xl font-secondary text-primary'>
                SaaS<span className='font-medium'>Candy</span>
              </span>
            </Link>
          </div>

          {/* User Information */}
          <div className='space-y-8'>
            {/* Profile Header */}
            <div className='text-center space-y-4'>
              {image ? (
                <div className='flex justify-center'>
                  <Link
                    href={image}
                    aria-label='Profile picture'
                    className='w-24 h-24 rounded-full border-4 border-orange-background'
                  />
                </div>
              ) : (
                <div className='flex justify-center'>
                  <div className='w-24 h-24 rounded-full bg-orange-background flex items-center justify-center text-white text-2xl font-bold'>
                    {name?.charAt(0) || email?.charAt(0) || 'U'}
                  </div>
                </div>
              )}

              <div>
                <Heading
                  as='h2'
                  size='md'
                  className='text-primary'
                  content={name || 'User'}
                />

                <Typography className='text-gray-background mt-1'>
                  {email}
                </Typography>
              </div>
            </div>

            {/* Account Information Cards */}
            <div className='space-y-4'>
              {/* Personal Information */}
              <div className='bg-blue-foreground opacity-80 rounded-lg p-6'>
                <Heading
                  as='h3'
                  size='md'
                  className='text-primary mb-4'
                  content='Personal Information'
                />

                <div className='space-y-3'>
                  <div className='flex justify-between items-center'>
                    <span className='text-blue-background font-bold'>
                      Name:
                    </span>
                    <span className='font-medium text-primary'>
                      {name || 'Not provided'}
                    </span>
                  </div>
                  <div className='flex justify-between items-center'>
                    <span className='text-blue-background font-bold'>
                      Email:
                    </span>
                    <span className='font-medium text-primary'>{email}</span>
                  </div>
                  <div className='flex justify-between items-center'>
                    <span className='text-blue-background font-bold'>
                      Account Type:
                    </span>
                    <span className='px-2 py-1 bg-blue-foreground text-blue-background rounded-full text-md'>
                      {image ? 'Social Account' : 'Email Account'}
                    </span>
                  </div>
                </div>
              </div>

              {/* Account Settings */}
              <div className='bg-blue-foreground opacity-80 rounded-lg p-6'>
                <Heading
                  as='h3'
                  size='md'
                  className='text-primary mb-4'
                  content='Account Settings'
                />

                <div className='space-y-3'>
                  <Button
                    variant='secondary'
                    className='w-full justify-start font-semibold'
                    disabled
                  >
                    Edit Profile (Coming Soon)
                  </Button>
                  <Button
                    variant='secondary'
                    className='w-full justify-start font-semibold'
                    disabled
                  >
                    Change Password (Coming Soon)
                  </Button>
                  <Button
                    variant='secondary'
                    className='w-full justify-start font-semibold'
                    disabled
                  >
                    Privacy Settings (Coming Soon)
                  </Button>
                </div>
              </div>
            </div>

            {/* Action Buttons */}
            <div className='flex gap-4 pt-6'>
              <Button
                variant='secondary'
                className='flex-1'
                onClick={handleGoBack}
              >
                Back
              </Button>
              <Button
                variant='tertiary'
                className='flex-1'
                onClick={handleSignOut}
              >
                Sign Out
              </Button>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}

export default AccountDetailPage;
