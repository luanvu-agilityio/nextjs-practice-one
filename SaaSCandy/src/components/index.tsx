export * from './common';
export * from './layout';
export * from './form';
export * from './pages';
export * from './NavBar';
export * from './SocialButton';
export * from './UserMenu';
export * from './FeaturedCard';
export * from './Share';
export * from './NewsletterSignup';
export * from './Tags';
export * from './BlogCard';
export * from './BlogList';
export * from './ContentRenderer';
export * from './PricingPlanCard';
export * from './Switch';
export * from './FAQSection';
export * from './HeroSection';
export * from './JoinUsSection';
