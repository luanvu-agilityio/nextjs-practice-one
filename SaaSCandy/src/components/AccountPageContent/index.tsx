'use client';

import { Fragment, useEffect, useState } from 'react';
import Link from 'next/link';
import { useSession, signOut } from '@/lib/auth-client';
import { useRouter } from 'next/navigation';
import Image from 'next/image';

// Components
import { Button, Heading, Typography } from '@/components/common';
import {
  Breadcrumb,
  BreadcrumbList,
  BreadcrumbItem,
  BreadcrumbLink,
  BreadcrumbSeparator,
  BreadcrumbPage,
} from '@/components/common/Breadcrumb';
import { Spinner } from '@/components';
import { showToast } from '@/components/common/Toast';
import EditProfileModal from '../EditProfileModal';
import ChangePasswordModal from '../ChangePasswordModal';

// Icons
import LogoIcon from '@/components/icons/Logo';

// Utils & Constants
import { ROUTES, TOAST_MESSAGES } from '@/constants';

// Types
import { TOAST_VARIANTS } from '@/types/toast';

function AccountPageContent() {
  const { data: session, isPending } = useSession();
  const router = useRouter();
  const [isEditProfileOpen, setIsEditProfileOpen] = useState(false);
  const [isChangePasswordOpen, setIsChangePasswordOpen] = useState(false);

  // Redirect if not authenticated
  useEffect(() => {
    if (!isPending && !session?.user) {
      router.push(ROUTES.SIGN_IN);
    }
  }, [router, session?.user, isPending]);

  if (isPending) {
    return (
      <div className='flex justify-center items-center min-h-screen'>
        <Spinner />
      </div>
    );
  }

  if (!session?.user) {
    return null;
  }

  const {
    user: { name, email, image },
  } = session;

  const handleGoBack = () => {
    router.back();
  };

  const handleSignOut = async () => {
    try {
      await signOut();
      router.push(ROUTES.HOME);
      router.refresh();
      showToast({
        ...TOAST_MESSAGES.ACCOUNT.SIGNOUT_SUCCESS,
        variant: TOAST_VARIANTS.SUCCESS,
      });
    } catch (error) {
      console.error('Sign out error:', error);
      showToast({
        ...TOAST_MESSAGES.ACCOUNT.SIGNOUT_ERROR,
        variant: TOAST_VARIANTS.ERROR,
      });
    }
  };

  const handleEditProfileSuccess = () => {
    showToast({
      ...TOAST_MESSAGES.ACCOUNT.PROFILE_UPDATE_SUCCESS,
      variant: TOAST_VARIANTS.SUCCESS,
    });
  };

  const handleChangePasswordSuccess = () => {
    showToast({
      ...TOAST_MESSAGES.ACCOUNT.PASSWORD_CHANGE_SUCCESS,
      variant: TOAST_VARIANTS.SUCCESS,
    });
  };

  const breadcrumbs = [
    { label: 'Home', href: ROUTES.HOME, isActive: false },
    { label: 'Account', href: ROUTES.ACCOUNT, isActive: true },
  ];

  const isSocialAccount = !!image;

  return (
    <>
      <div className='flex flex-col items-center max-w-[1296px] mx-auto min-h-screen py-4 sm:py-8 px-4 sm:px-6'>
        {/* Header Section  */}
        <div className='pb-6 sm:pb-12 w-full'>
          <Heading
            as='h1'
            size='xl'
            content='Account Details'
            className='text-center text-2xl sm:text-3xl lg:text-4xl'
          />

          {/* Breadcrumb */}
          <Breadcrumb className='justify-center mt-2 sm:mt-4'>
            <BreadcrumbList>
              {breadcrumbs.map((item, index) => (
                <Fragment key={item.label}>
                  <BreadcrumbItem>
                    {item.isActive ? (
                      <BreadcrumbPage className='text-sm sm:text-xs'>
                        {item.label}
                      </BreadcrumbPage>
                    ) : (
                      <BreadcrumbLink
                        href={item.href}
                        className='text-sm sm:text-xs'
                      >
                        {item.label}
                      </BreadcrumbLink>
                    )}
                  </BreadcrumbItem>
                  {index < breadcrumbs.length - 1 && <BreadcrumbSeparator />}
                </Fragment>
              ))}
            </BreadcrumbList>
          </Breadcrumb>
        </div>

        {/* Account Detail Section  */}
        <div className='py-8 px-4 sm:py-14 sm:px-16 w-full lg:max-w-159 rounded-2xl sm:rounded-4xl border border-form-border-color shadow-form bg-white'>
          <div className='flex flex-col gap-6 sm:gap-10'>
            {/* Logo */}
            <div className='flex justify-center'>
              <Link
                href={ROUTES.HOME}
                className='flex items-center gap-2 sm:gap-3'
              >
                <LogoIcon className='w-8 h-8 sm:w-10 sm:h-10' />
                <span className='text-2xl sm:text-3xl font-secondary text-primary'>
                  SaaS<span className='font-medium'>Candy</span>
                </span>
              </Link>
            </div>

            {/* User Information */}
            <div className='space-y-6 sm:space-y-8'>
              {/* Profile Header  */}
              <div className='text-center space-y-3 sm:space-y-4'>
                {image ? (
                  <div className='flex justify-center'>
                    <Image
                      src={image}
                      alt='Profile picture'
                      width={96}
                      height={96}
                      className='w-20 h-20 sm:w-24 sm:h-24 rounded-full border-4 border-orange-background object-cover'
                    />
                  </div>
                ) : (
                  <div className='flex justify-center'>
                    <div className='w-20 h-20 sm:w-24 sm:h-24 rounded-full bg-orange-background flex items-center justify-center text-white text-xl sm:text-2xl font-bold'>
                      {name?.charAt(0) || email?.charAt(0) || 'U'}
                    </div>
                  </div>
                )}

                <div>
                  <Heading
                    as='h2'
                    size='md'
                    className='text-primary text-xl sm:text-2xl'
                    content={name || 'User'}
                  />

                  <Typography className='text-gray-background mt-1 text-sm sm:text-xs break-all'>
                    {email}
                  </Typography>
                </div>
              </div>

              {/* Account Information Cards */}
              <div className='space-y-3 sm:space-y-4'>
                {/* Personal Information */}
                <div className='bg-blue-foreground opacity-80 rounded-lg p-4 sm:p-6'>
                  <Heading
                    as='h3'
                    size='md'
                    className='text-primary mb-3 sm:mb-4 text-lg sm:text-xl'
                    content='Personal Information'
                  />

                  <div className='space-y-2 sm:space-y-3'>
                    <div className='flex flex-col sm:flex-row sm:justify-between sm:items-center gap-1 sm:gap-0  sm:text-lg'>
                      <span className='text-blue-background font-bold'>
                        Name:
                      </span>
                      <span className='font-medium text-primary'>
                        {name || 'Not provided'}
                      </span>
                    </div>
                    <div className='flex flex-col sm:flex-row sm:justify-between sm:items-center gap-1 sm:gap-0  sm:text-lg'>
                      <span className='text-blue-background font-bold'>
                        Email:
                      </span>
                      <span className='font-medium text-primary break-all'>
                        {email}
                      </span>
                    </div>
                    <div className='flex flex-col sm:flex-row sm:justify-between sm:items-center gap-1 sm:gap-0 sm:text-lg'>
                      <span className='text-blue-background font-bold'>
                        Account Type:
                      </span>
                      <span className='px-2 py-1 bg-blue-foreground text-blue-background rounded-full text-sm sm:text-md inline-block w-fit'>
                        {isSocialAccount ? 'Social Account' : 'Email Account'}
                      </span>
                    </div>
                  </div>
                </div>

                {/* Account Settings */}
                <div className='bg-blue-foreground opacity-80 rounded-lg p-4 sm:p-6'>
                  <Heading
                    as='h3'
                    size='md'
                    className='text-primary mb-3 sm:mb-4 text-lg sm:text-xl'
                    content='Account Settings'
                  />

                  <div className='space-y-2 sm:space-y-3'>
                    <Button
                      variant='secondary'
                      className='w-full justify-start font-semibold sm:text-lg'
                      onClick={() => setIsEditProfileOpen(true)}
                    >
                      Edit Profile
                    </Button>

                    <Button
                      variant='secondary'
                      className='w-full justify-start font-semibold  sm:text-lg'
                      onClick={() => setIsChangePasswordOpen(true)}
                      disabled={isSocialAccount}
                    >
                      <span className='block sm:inline'>Change Password</span>
                      {isSocialAccount && (
                        <span className='block sm:inline text-xs sm:text-sm ml-0 sm:ml-1 mt-1 sm:mt-0'>
                          (Not available for social accounts)
                        </span>
                      )}
                    </Button>

                    <Button
                      variant='secondary'
                      className='w-full justify-start font-semibold  sm:text-lg'
                      disabled
                    >
                      Privacy Settings (Coming Soon)
                    </Button>
                  </div>
                </div>
              </div>

              {/* Action Buttons  */}
              <div className='flex flex-col sm:flex-row gap-3 sm:gap-4 pt-4 sm:pt-6'>
                <Button
                  variant='secondary'
                  className='flex-1 w-full  sm:text-lg'
                  onClick={handleGoBack}
                >
                  Back
                </Button>
                <Button
                  variant='tertiary'
                  className='flex-1 w-full  sm:text-lg'
                  onClick={handleSignOut}
                >
                  Sign Out
                </Button>
              </div>
            </div>
          </div>
        </div>
      </div>

      {/* Modals */}
      <EditProfileModal
        open={isEditProfileOpen}
        onOpenChange={setIsEditProfileOpen}
        onSuccess={handleEditProfileSuccess}
      />

      <ChangePasswordModal
        open={isChangePasswordOpen}
        onOpenChange={setIsChangePasswordOpen}
        onSuccess={handleChangePasswordSuccess}
      />
    </>
  );
}

export default AccountPageContent;
