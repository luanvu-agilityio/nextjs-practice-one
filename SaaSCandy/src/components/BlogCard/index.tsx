'use client';

import React from 'react';
import Image from 'next/image';
import Link from 'next/link';

// Components
import { Heading } from '@/components/common';

interface BlogCardProps {
  slug: string;
  title: string;
  image?: string;
  date: string;
}

function BlogCard({ slug, title, image, date }: Readonly<BlogCardProps>) {
  return (
    <article className='group'>
      <Link
        href={`/blog/${slug}`}
        className='block rounded-lg overflow-hidden hover:shadow-md transition-shadow bg-white'
      >
        <div className='w-full overflow-hidden'>
          {image ? (
            <Image
              alt={title}
              className='object-cover w-full h-full group-hover:scale-105 transition-transform duration-300'
              src={image}
              width={416}
              height={284}
            />
          ) : (
            <div className='w-full h-full flex items-center justify-center text-gray-background'>
              No image
            </div>
          )}
        </div>

        <div className='flex flex-col gap-4 py-10'>
          <Heading
            as='h3'
            size='md'
            content={title}
            className='text-primary font-medium text-xl'
          />
          <div className='text-xs text-gray-background'>{date}</div>
        </div>
      </Link>
    </article>
  );
}
export { BlogCard };
