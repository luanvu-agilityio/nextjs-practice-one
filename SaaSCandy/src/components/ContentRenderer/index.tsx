import { ContentBlock } from '@/types/blog-post';
import { Heading, Typography } from '@/components/common';
import Image from 'next/image';

interface ContentRendererProps {
  blocks: ContentBlock[];
  className?: string;
}

function ContentRenderer({
  blocks,
  className = '',
}: Readonly<ContentRendererProps>) {
  const renderBlock = (block: ContentBlock) => {
    switch (block.type) {
      case 'heading':
        return (
          <Heading
            key={block.id}
            as={`h${block.content.level}`}
            size={block.content.level === 2 ? 'xl' : 'lg'}
            content={block.content.text}
            className='text-gray-900 font-bold mb-3 sm:mb-4 text-xl sm:text-2xl lg:text-3xl'
          />
        );

      case 'paragraph':
        return (
          <Typography
            key={block.id}
            content={block.content.text}
            className='text-gray-600 mb-3 sm:mb-4 leading-relaxed text-sm sm:text-xs lg:text-lg'
          />
        );

      case 'list': {
        const ListTag = block.content.ordered ? 'ol' : 'ul';
        return (
          <ListTag
            key={block.id}
            className='mb-3 sm:mb-4 pl-4 sm:pl-6 space-y-1 sm:space-y-2'
          >
            {block.content.items.map((item: string, index: number) => (
              <li key={index} className='text-gray-600 text-sm sm:text-xs'>
                {item}
              </li>
            ))}
          </ListTag>
        );
      }

      case 'quote':
        return (
          <blockquote
            key={block.id}
            className='border-l-4 border-primary pl-3 sm:pl-4 py-2 mb-3 sm:mb-4 italic text-gray-700 bg-gray-50 rounded-r'
          >
            <Typography
              content={`"${block.content.text}"`}
              className='text-sm sm:text-xs'
            />
            {block.content.author && (
              <cite className='text-xs sm:text-sm text-gray-500'>
                â€” {block.content.author}
              </cite>
            )}
          </blockquote>
        );

      case 'image':
        return (
          <div key={block.id} className='mb-4 sm:mb-6'>
            <Image
              src={block.content.src}
              alt={block.content.alt}
              width={block.content.width || 800}
              height={block.content.height || 400}
              className='w-full rounded-lg'
            />
            {block.content.caption && (
              <Typography
                content={block.content.caption}
                className='text-xs sm:text-sm text-gray-500 text-center mt-2 italic'
              />
            )}
          </div>
        );

      case 'code':
        return (
          <pre
            key={block.id}
            className='bg-gray-900 text-gray-100 p-3 sm:p-4 rounded-lg mb-3 sm:mb-4 overflow-x-auto text-xs sm:text-sm'
          >
            <code
              className={
                block.content.language
                  ? `language-${block.content.language}`
                  : undefined
              }
            >
              {block.content.code}
            </code>
          </pre>
        );

      default:
        const _exhaustiveCheck: never = block;
        return _exhaustiveCheck;
    }
  };

  return (
    <div
      className={`prose prose-sm sm:prose-base lg:prose-lg max-w-none ${className}`}
    >
      {blocks.map(renderBlock)}
    </div>
  );
}

export { ContentRenderer };
