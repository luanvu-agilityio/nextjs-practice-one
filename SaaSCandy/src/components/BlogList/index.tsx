import { getAllPosts } from '@/helpers';
import { BlogCard } from '../BlogCard';

async function BlogList() {
  const posts = await getAllPosts();
  return (
    <>
      {posts.map(p => (
        <BlogCard key={p.slug} {...p} />
      ))}
    </>
  );
}

export default BlogList;
