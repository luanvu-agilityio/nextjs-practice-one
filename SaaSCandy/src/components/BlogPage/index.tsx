// Components
import Section from '../common/Section';
import { BlogCard } from '../BlogCard';

// Helper
import { getAllPosts } from '@/helpers';

async function BlogPage() {
  const posts = await getAllPosts();

  return (
    <Section className='bg-white'>
      <div className='grid sm:grid-cols-2 lg:grid-cols-3 gap-6'>
        {posts.map(p => (
          <BlogCard key={p.slug} {...p} />
        ))}
      </div>
    </Section>
  );
}
export default BlogPage;
