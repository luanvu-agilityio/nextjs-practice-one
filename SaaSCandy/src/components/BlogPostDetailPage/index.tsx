import Image from 'next/image';
import Link from 'next/link';

// Components
import { Heading } from '../common';
import Section from '../common/Section';
import { Tags } from '../Tags';
import { NewsletterSignup } from '../NewsletterSignup';
import { Share } from '../Share';
import { ContentRenderer } from '../ContentRenderer';

// Types
import { BlogPost } from '@/types/blog-post';

function BlogPostDetailPage({ post }: { post: Readonly<BlogPost> }) {
  return (
    <Section className='bg-white'>
      {/* Post Meta - Top */}
      <div className='flex items-center gap-4 text-sm text-gray-background mb-7'>
        <span>{post.date}</span>
        <span>{post.comments} Comments</span>
      </div>

      {/* Post Title */}
      <div className='flex items-center justify-between gap-6 mb-6'>
        <Heading as='h2' size='xl' className='font-bold' content={post.title} />
        {post.author && (
          <div className='flex items-center justify-end gap-6 w-76.5'>
            {post.author.avatar && (
              <Image
                src={post.author.avatar}
                alt={post.author.name}
                width={80}
                height={80}
                className='rounded-full'
              />
            )}
            <div className='flex flex-col'>
              <span className='text-xl font-bold '>{post.author.name}</span>
              <span className='text-xl font-regular'>Author</span>
            </div>
          </div>
        )}
      </div>
      {/* Featured Image */}
      {post.image && (
        <div className='mb-8'>
          <Image
            src={post.image}
            alt={post.title}
            width={1296}
            height={440}
            className='w-full  object-cover rounded-3xl'
          />
        </div>
      )}
      <div className='flex items-start justify-between gap-6'>
        <article className='flex-1'>
          {/* Post Content */}
          {post.contentType === 'structured' && post.contentBlocks ? (
            <ContentRenderer blocks={post.contentBlocks} className='mb-8' />
          ) : (
            <div
              className='prose prose-lg max-w-none mb-8'
              dangerouslySetInnerHTML={{ __html: post.content || '' }}
            />
          )}

          {/* Tags Section */}
          <div className='border-t pt-8 mb-8'>
            <Heading
              as='h3'
              className='text-2xl font-bold text-gray-900 mb-4'
              content='Tags'
            />
            <Tags tags={post?.tags ?? []} />
          </div>

          {/* Share Section */}
          <div className='border-t pt-8 mb-8'>
            <Heading
              as='h3'
              className='text-2xl font-bold text-gray-900 mb-4'
              content='Share'
            />
            <ul className='space-y-2'>
              <li>
                <Link
                  href={`https://www.facebook.com/sharer/sharer.php?u=${encodeURIComponent(typeof window !== 'undefined' ? window.location.href : '')}`}
                  target='_blank'
                  rel='noopener noreferrer'
                  className='text-gray-600 hover:text-gray-900 transition-colors'
                >
                  Facebook
                </Link>
              </li>
              <li>
                <Link
                  href={`https://twitter.com/intent/tweet?url=${encodeURIComponent(typeof window !== 'undefined' ? window.location.href : '')}&text=${encodeURIComponent(post.title)}`}
                  target='_blank'
                  rel='noopener noreferrer'
                  className='text-gray-600 hover:text-gray-900 transition-colors'
                >
                  Twitter
                </Link>
              </li>
              <li>
                <Link
                  href={`https://www.linkedin.com/sharing/share-offsite/?url=${encodeURIComponent(typeof window !== 'undefined' ? window.location.href : '')}`}
                  target='_blank'
                  rel='noopener noreferrer'
                  className='text-gray-600 hover:text-gray-900 transition-colors'
                >
                  LinkedIn
                </Link>
              </li>
            </ul>
          </div>

          {/* Newsletter Section */}
          <div className='border-t pt-8'>
            <NewsletterSignup />
          </div>
        </article>
        {/* Sidebar */}
        <div className='sticky w-104 py-12 px-11 space-y-12 shadow-form rounded-lg '>
          <Share />
          <NewsletterSignup className='w-full' />
        </div>
      </div>
    </Section>
  );
}
export default BlogPostDetailPage;
