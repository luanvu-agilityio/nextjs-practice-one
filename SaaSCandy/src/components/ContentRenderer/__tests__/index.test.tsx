import { render } from '@testing-library/react';
import { ContentRenderer } from '@/components/ContentRenderer';
import { ContentBlock } from '@/types/blog-post';

const mockBlocks: ContentBlock[] = [
  {
    id: '1',
    type: 'heading',
    content: { level: 2, text: 'Test Heading' },
  },
  {
    id: '2',
    type: 'paragraph',
    content: { text: 'Test paragraph content' },
  },
  {
    id: '3',
    type: 'list',
    content: { ordered: false, items: ['Item 1', 'Item 2', 'Item 3'] },
  },
  {
    id: '4',
    type: 'quote',
    content: { text: 'Test quote', author: 'Test Author' },
  },
  {
    id: '5',
    type: 'image',
    content: {
      src: '/test.jpg',
      alt: 'Test image',
      width: 800,
      height: 400,
      caption: 'Test caption',
    },
  },
  {
    id: '6',
    type: 'code',
    content: { code: 'const test = "hello";', language: 'javascript' },
  },
];

describe('ContentRenderer - Snapshot Tests', () => {
  it('should match snapshot with all block types', () => {
    const { container } = render(<ContentRenderer blocks={mockBlocks} />);
    expect(container).toMatchSnapshot();
  });

  it('should render all content blocks', () => {
    const { container } = render(<ContentRenderer blocks={mockBlocks} />);

    expect(container.querySelector('h2')).toBeInTheDocument();
    expect(container.querySelector('ul')).toBeInTheDocument();
    expect(container.querySelector('blockquote')).toBeInTheDocument();
    expect(container.querySelector('img')).toBeInTheDocument();
    expect(container.querySelector('pre')).toBeInTheDocument();
  });

  it('should apply custom className', () => {
    const { container } = render(
      <ContentRenderer blocks={mockBlocks} className='custom-class' />
    );

    const wrapper = container.firstChild as HTMLElement;
    expect(wrapper).toHaveClass('custom-class');
  });
});
