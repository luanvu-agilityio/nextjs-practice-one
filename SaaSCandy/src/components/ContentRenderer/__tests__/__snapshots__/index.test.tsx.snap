// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ContentRenderer - Snapshot Tests should match snapshot with all block types 1`] = `
<div>
  <div
    class="prose prose-sm sm:prose-base lg:prose-lg max-w-none "
  >
    <h2
      aria-level="2"
      class="text-gray-900 font-bold mb-3 sm:mb-4 text-xl sm:text-2xl lg:text-3xl"
      role="heading"
    >
      Test Heading
    </h2>
    <p
      class="text-gray-600 mb-3 sm:mb-4 text-sm sm:text-xs lg:text-lg"
      role="paragraph"
    >
      Test paragraph content
    </p>
    <ul
      class="mb-3 sm:mb-4 pl-4 sm:pl-6 space-y-1 sm:space-y-2"
    >
      <li
        class="text-gray-600 text-sm sm:text-xs"
      >
        Item 1
      </li>
      <li
        class="text-gray-600 text-sm sm:text-xs"
      >
        Item 2
      </li>
      <li
        class="text-gray-600 text-sm sm:text-xs"
      >
        Item 3
      </li>
    </ul>
    <blockquote
      class="border-l-4 border-primary pl-3 sm:pl-4 py-2 mb-3 sm:mb-4 italic text-gray-700 bg-gray-50 rounded-r"
    >
      <p
        class="text-sm sm:text-xs"
        role="paragraph"
      >
        "Test quote"
      </p>
      <cite
        class="text-xs sm:text-sm text-gray-500"
      >
        â€” 
        Test Author
      </cite>
    </blockquote>
    <div
      class="mb-4 sm:mb-6"
    >
      <img
        alt="Test image"
        class="w-full rounded-lg"
        data-nimg="1"
        decoding="async"
        height="400"
        loading="lazy"
        src="/_next/image?url=%2Ftest.jpg&w=1920&q=75"
        srcset="/_next/image?url=%2Ftest.jpg&w=828&q=75 1x, /_next/image?url=%2Ftest.jpg&w=1920&q=75 2x"
        style="color: transparent;"
        width="800"
      />
      <p
        class="text-xs sm:text-sm text-gray-500 text-center mt-2 italic"
        role="paragraph"
      >
        Test caption
      </p>
    </div>
    <pre
      class="bg-gray-900 text-gray-100 p-3 sm:p-4 rounded-lg mb-3 sm:mb-4 overflow-x-auto text-xs sm:text-sm"
    >
      <code
        class="language-javascript"
      >
        const test = "hello";
      </code>
    </pre>
  </div>
</div>
`;
